client/jetpack-connect/controller.js

```diff
@@ -186,6 +181,11 @@ export function connect( context, next ) {
 export function signupForm( context, next ) {
 	analytics.pageView.record( 'jetpack/connect/authorize', 'Jetpack Authorize' );
 
+	if ( retrieveMobileRedirect() && ! userModule.get() ) {
+		// Force login for mobile app flow. App will this request and prompt native login.
+		return window.location.replace( login( { isNative: true, redirectTo: context.path } ) );
```

Comment:
Now making an explicit server request, which is easier for mobile app to intercept. Request will never actually hit the server.